enterprise/backup/src/test/java/org/neo4j/commandline/admin/BackupUsageTest.java

```diff
@@ -0,0 +1,109 @@
+import java.io.UnsupportedEncodingException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.neo4j.backup.impl.BackupHelpOutput;
+import org.neo4j.backup.impl.OnlineBackupCommandProvider;
+import org.neo4j.backup.impl.ParametrisedOutsideWorld;
+import org.neo4j.io.fs.DefaultFileSystemAbstraction;
+import org.neo4j.io.fs.FileSystemAbstraction;
+
+import static org.junit.Assert.assertEquals;
+
+public class BackupUsageTest
+{
+    OnlineBackupCommandProvider onlineBackupCommandProvider = new OnlineBackupCommandProvider();
+    CommandLocator commandLocator = AugmentedCommandLocator.fromFixedArray( onlineBackupCommandProvider );
+
+    @Test
+    public void outputMatchesExpectedForMissingBackupDir() throws UnsupportedEncodingException
+    {
+        // when
+        String output = runBackup();
+
+        // then
+        String reason = "Missing argument 'backup-dir'\n\n";
+        assertEquals( reason + BackupHelpOutput.BACKUP_OUTPUT, output );
+    }
+
+    @Test
+    public void missingBackupName() throws UnsupportedEncodingException
+    {
+        // when
+        String output = runBackup( "--backup-dir=target" );
+
+        // then
+        String reason = "Missing argument 'name'\n\n";
+        assertEquals( reason + BackupHelpOutput.BACKUP_OUTPUT, output );
+    }
+
+    @Test
+    public void incorrectBackupDirectory() throws IOException
+    {
+        // when
+        Path backupDirectoryResolved = new File( "." ).toPath().toRealPath().resolve( "non_existing_dir" );
+        String output = runBackup( "--backup-dir=non_existing_dir", "--name=mybackup" );
+
+        // then
+        String reason = String.format( "command failed: Directory '%s' does not exist.\n", backupDirectoryResolved.toFile().toString() );
+        assertEquals( reason, output );
+    }
+
+    private String runBackup( String... args ) throws UnsupportedEncodingException
+    {
+        return runBackup( false, args );
+    }
+
+    private String runBackup( boolean debug, String... args ) throws UnsupportedEncodingException
+    {
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        PrintStream printStream = new PrintStream( baos, true, "utf-8" );
+        ParametrisedOutsideWorld outsideWorld = capturableOutputOutsideWorld( printStream );
+        AdminTool subject = new AdminTool( commandLocator, AugmentedBlockerLocator.fromList(), outsideWorld, debug );
+        Path homeDir = new File( "." ).toPath();
+        Path configDir = new File( "." ).toPath();
+        List<String> params = new ArrayList();
+        params.add( "backup" );
+        params.addAll( Arrays.asList( args ) );
+        String[] argArray = params.toArray( new String[params.size()] );
+        subject.execute( homeDir, configDir, argArray );
+        return baos.toString();
+    }
+
+    private ParametrisedOutsideWorld capturableOutputOutsideWorld( PrintStream printStream )
```

Comment:
There exists a JUnit rule that captures `System.out` - `SuppressOutput`. Can it be used instead?